config.go

```diff
@@ -10,6 +10,13 @@ import (
 	"github.com/segmentio/parquet-go/compress"
 )
 
+type PageReadMode int
+
+const (
+	PageReadModeAsync PageReadMode = iota
+	PageReadModeSync
+)
```

Comment:
An alternative model I had considered would be to expose this customization as an interface, for example:

```go
type PageReadMode interface {
    PageReaders(columns []ColumnChunk) ([]Pages, error)
}
```
The main upside I would imagine is that it allows the application to experiment with other strategies if needed, potentially even compose them.

We need to account for the support for asynchronous cancellation in async mode, so maybe the signature could be:
```go
type PageReadMode interface {
    PageReaders(ctx context.Context, columns []ColumnChunk) ([]Pages, error)
}
```

I'm not entirely convinced that it is a better model, but I wanted to bring it up for discussion. Let me know what you think about it.

Reply:
I think this could be interesting! This question might already test the way people might abuse this or if it works as intended: do you imagine that this could be used to implement (custom) page caching?

Reply:
This does seem to be a question of how much flexibility we'd like to provide at this point. The idea of routing page requests to different places is tempting, but perhaps should be designed explicitly instead of added here.

I wonder if it would better handled through the Page interface? That way if you are accessing the pages of a column individually (instead of entire rows at a time) you would still have access to some kind of flexible routing behavior?

To me this feels like a specific configuration of the behavior of `rowGroupRows` then something generic.

Reply:
OK, your call @joe-elliott, I'm fine keeping it as-is and discussing further later on 👍

Reply:
sounds good. let's leave it as is then and maybe discuss how to work this in more holistically.

Reply:
Works for me.